geo/src/algorithm/intersects/line_string.rs

```diff
@@ -3,31 +3,141 @@ use crate::BoundingRect;
 use crate::*;
 
 // Blanket implementation using self.lines().any().
-impl<T, G> Intersects<G> for LineString<T>
+macro_rules! blanket_intersects_linestring {
+    ($t:ty) => {
+        impl<T> $crate::Intersects<$t> for LineString<T>
+        where
+            T: GeoNum,
+            Line<T>: Intersects<$t>,
+            $t: BoundingRect<T>,
+        {
+            fn intersects(&self, rhs: &$t) -> bool {
+                if has_disjoint_bboxes(self, rhs) {
+                    return false;
+                }
+                self.lines().any(|l| l.intersects(rhs))
+            }
+        }
+    };
+}
+
+blanket_intersects_linestring!(Coord<T>);
+blanket_intersects_linestring!(Point<T>);
+blanket_intersects_linestring!(MultiPoint<T>);
+
+blanket_intersects_linestring!(Line<T>);
+blanket_intersects_linestring!(LineString<T>);
+symmetric_intersects_impl!(LineString<T>, MultiLineString<T>);
+
+impl<T> Intersects<Polygon<T>> for LineString<T>
+where
+    T: GeoNum,
+    Line<T>: Intersects<Line<T>>,
```

Comment:
Aren't these `Intersects` constraints a (redundant) artifact from the blanket impl?